### PR TITLE
Adds Dynamic Button Creation Functionality

### PR DESCRIPTION
**Pull request by Jcobo314**
This patch allows a user to create buttons with data based on the title of each column in the table. When a button is clicked, the first column of the table will always populate with the current time-stamp of the video being played, then each subsequent column will populate based on data assigned to a button.

Testing Steps:

  1. Run the application and clear all sessions
  2. Created a new session and name it "test"
  3. Click "File" -> "Load video file" then add a video to the application
  4. Click "+" in the "Coding Assistance Buttons" panel
  5. Name this button "button1", assign the hotkey "a",  leave the "Time" field blank and populate each field with data of your choice then click "Create". Do not close the dialog
  6. Repeat step 5 to create a second button with a different name, hotkey, and data. Close the dialog
  7. Press "a" on your keyboard and ensure the table populates with button1's data.
  8. Repeat step 7 with the second button's hotkey. Ensure the data populates and is different from button1's
  9. Click "-" in the "Coding Assistance Buttons" panel
  10. Delete "button1"
  11. Press "a" and ensure no data populates in the table
  12. Close the application

Type: New Functionality

### CODE REVIEW

File: Controllers/window_controller.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def __init__(self, window):
         self._window = window
 
         self._manager = Manager()
+        self.encoding_buttons = []
```
Comment (Briggs-Richardson) [inline Controllers/window_controller.py]: I don't think this is being used. If not, then remove.

File: Controllers/window_controller.py [hunk 2]
```diff
@@ -443,17 +444,20 @@ def create_coding_assistance_button(self):
...
         button_name = self.add_coding_assistance_button_dialog.apply_text_field.text()
         button_hotkey = self.add_coding_assistance_button_dialog.hotkey_field.text()
+        data = []
+        for text in self.add_coding_assistance_button_dialog.dynamic_line_edits:
+            data.append(text.text())
         new_button = QPushButton(button_name)
-        new_button.setObjectName(button_name)
         new_button.setShortcut(QKeySequence(button_hotkey))
-        new_button_definition = ButtonDefinition(new_button, button_name, button_hotkey)
+        self.encoding_buttons.append(new_button)
```
Comment (Briggs-Richardson) [inline Controllers/window_controller.py]: If this serves no purpose, then remove.

File: View/add_coding_assistance_button_dialog.py [hunk 3]
```diff
@@ -3,12 +3,14 @@
 
 class AddCodingAssistanceButtonDialog(QDialog):
 
-    def __init__(self):
+    def __init__(self, table):
         """
         Constructor: Initializes the layout of the Add Coding Assistance Button dialog
```
Comment (Briggs-Richardson) [inline View/add_coding_assistance_button_dialog.py:8]: Add a parameters comment, describing what the table parameter is.

File: View/button_panel.py [hunk 4]
```diff
@@ -74,10 +74,9 @@ def create_coding_assistance_button(self, button_definition):
         """Adds a new_button to the Coding Assistance Panel"""
         self.button_layout.addWidget(button_definition.button)
 
-
     def delete_coding_assistance_button(self, button_name):
         """Deletes a button in the Coding Assistance Panel"""
```
Comment (Briggs-Richardson) [inline View/button_panel.py]: Add a parameters comment, describing what the button_name is.

File: Application/button_definition.py [hunk 5]
```diff
@@ -6,10 +6,11 @@ class ButtonDefinition:
     An object defining the attributes for a button within the coding assistance
     panel.
     """
-    def __init__(self, button, button_id, hotkey):
+    def __init__(self, button, button_id, hotkey, data):
         """
         Constructor - Creates an instance of ButtonDefinition
```
Comment (Briggs-Richardson) [inline Application/button_definition.py:8]: Add a parameters comment, describing what the button, button_id, hotkey, and data are.

File: Controllers/window_controller.py [hunk 6]
```diff
@@ -463,26 +467,34 @@ def create_coding_assistance_button(self):
...
 
-    def dynamic_button_click(self):
-        """Add dummy data to table"""
-        date_time = datetime.now()
-        dt_string = date_time.strftime("%d/%m/%Y %H:%M:%S")
-
-        column = 0
+    def dynamic_button_click(self, button_definition):
+        """
+        Add button data to table when clicked
```
Comment (Briggs-Richardson) [inline Controllers/window_controller.py:482]: Add a parameters comment, describing what the button_definition is

File: Controllers/window_controller.py [hunk 7]
```diff
@@ -463,26 +467,34 @@ def create_coding_assistance_button(self):
...
                     self._manager.hotkey_list.remove(button_definition.hotkey)
                 self._window.coding_assistance_panel.button_panel.delete_coding_assistance_button(check_button_name)
 
-    def dynamic_button_click(self):
-        """Add dummy data to table"""
-        date_time = datetime.now()
-        dt_string = date_time.strftime("%d/%m/%Y %H:%M:%S")
-
-        column = 0
+    def dynamic_button_click(self, button_definition):
```
Comment (Briggs-Richardson) [inline Controllers/window_controller.py:480]: Add a slot annotation above the method: `@Slot(ButtonDefinition)`

**Review by Briggs-Richardson — CHANGES_REQUESTED** [review]
First CR

File: Controllers/window_controller.py (hunk 7 above)
Comment (Jcobo314) [reply Controllers/window_controller.py:480]: This isn't a slot function because it's being called within window_controller

**Review by Jcobo314 — COMMENTED** [review]
Testing Step 4 and 7: When creating the new session with the table title set in the session creation page, the title for the table is not updated when the session opens. Could be a future bug fixing task.
Otherwise all testing steps for the new functionality passed

**Pull request merged**